app/controllers/placements/support/provider_suggestions_controller.rb

```diff
@@ -0,0 +1,33 @@
+class Placements::Support::ProviderSuggestionsController < Placements::Support::ApplicationController
+  def index
+    accredited_providers = AccreditedProviderApi.call
+    filtered_providers = filter_providers(accredited_providers, params[:query])
+    providers =
+      filtered_providers.map { |provider| formatted_provider(provider) }
+    render json: providers
+  end
+
+  private
+
+  def filter_providers(providers, query)
+    return providers if query.blank?
+
+    downcase_query = query.downcase
+    providers.select do |provider|
+      [
+        provider["attributes"]["name"],
+        provider["attributes"]["postcode"],
+        provider["attributes"]["urn"],
+        provider["attributes"]["ukprn"]
+      ].any? { |attribute| attribute&.downcase&.include?(downcase_query) }
+    end
+  end
+
+  def formatted_provider(provider)
+    {
+      id: provider["id"],
+      name: provider["attributes"]["name"],
+      code: provider["attributes"]["code"]
+    }
+  end
+end
```

Comment:
This is more of a personal opinion but I think we should use {}.fetch here as it offers more descriptive errors when keys are missing, also it doesn't fail silently if the keys are missing. It's not really a blocker though.

Reply:
resolved in [abdfae3](https://github.com/DFE-Digital/itt-mentor-services/pull/48/commits/abdfae3e4ab05b2c4cf0a45a0c634fec1fbf6545)